app/controllers/ncr/proposals_controller.rb

```diff
@@ -29,6 +32,46 @@ def create
       end
     end
 
+    def not_approved
+      cart = Cart.find(params[:id])
+      if cart.approved?
+        redirect_to new_ncr_proposal_path, :alert => "That proposal's already approved. New proposal?"
+      end
+    end
+    def cart_owner
+      cart = Cart.find(params[:id])
+      if cart.requester != current_user
+        redirect_to new_ncr_proposal_path, :alert => 'You cannot restart that proposal'
```

Comment:
This is a good start, but we should really have a more comprehensive authorization layer sooner or later.

https://www.ruby-toolbox.com/categories/rails_authorization